api/src/main/java/com/code4ro/nextdoor/emergency/contact/service/impl/EmergencyContactServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+package com.code4ro.nextdoor.emergency.contact.service.impl;
+
+import com.code4ro.nextdoor.emergency.contact.dto.EmergencyContactDto;
+import com.code4ro.nextdoor.emergency.contact.entity.EmergencyContact;
+import com.code4ro.nextdoor.emergency.contact.repository.EmergencyContactRepository;
+import com.code4ro.nextdoor.emergency.contact.service.EmergencyContactService;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Service;
+
+@Service
+public class EmergencyContactServiceImpl implements EmergencyContactService {
+
+    private final EmergencyContactRepository emergencyContactRepository;
+
+    public EmergencyContactServiceImpl(EmergencyContactRepository emergencyContactRepository) {
+        this.emergencyContactRepository = emergencyContactRepository;
+    }
+
+    @Override
+    public EmergencyContactDto save(EmergencyContactDto emergencyContactDto) {
+        EmergencyContact emergencyContact = mapDtoToEntity(emergencyContactDto);
+        EmergencyContact emergencyContactDB = emergencyContactRepository.save(emergencyContact);
+
+        return mapDtoToDTO(emergencyContactDB);
+    }
+
+    @Override
+    public EmergencyContactDto update(EmergencyContactDto emergencyContactDto) {
+        Optional<EmergencyContact> dbEntity = emergencyContactRepository.findById(
+            UUID.fromString(emergencyContactDto.getId()));
+
+        if (dbEntity.isPresent()) {
+            EmergencyContact dbEmergencyContact = dbEntity.get();
+            dbEmergencyContact.setAddress(emergencyContactDto.getAddress());
+            dbEmergencyContact.setEmail(emergencyContactDto.getEmail());
+            dbEmergencyContact.setName(emergencyContactDto.getName());
+            dbEmergencyContact.setSurname(emergencyContactDto.getSurname());
+            dbEmergencyContact.setTelephoneNumber(emergencyContactDto.getTelephoneNumber());
+            return mapDtoToDTO(emergencyContactRepository.save(dbEmergencyContact));
+        } else {
+            return save(emergencyContactDto);
+        }
+    }
+
+    @Override
+    public void deleteById(String id) {
+        Optional<EmergencyContact> dbEntity = emergencyContactRepository.findById(
+            UUID.fromString(id));
+
+        if (dbEntity.isPresent()) {
+            emergencyContactRepository.deleteById(UUID.fromString(id));
```

Comment:
Aici ai putea sa apelezi direct delete, chiar daca nu exista informatia in baza de date ( in cazul in care nu exista nu se intampla nimic ). Asa faci 2 call-uri in baza de date.

Reply:
Da, nu se intampla nimic cat timp e un valid UUID. Poate ar fi buna o validare a lui in controller in acest caz.

Reply:
Dar chiar si daca e invalid nu ar trebui sa se intample nimic. Pentru ca e ‘DELETE FROM table WHERE uuid=“uuidPrimit”’ si daca nu exista nu executa delete

Reply:
dap, I was overengineering this :))